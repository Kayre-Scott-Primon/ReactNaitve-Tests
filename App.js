import React, { useState } from 'react'
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    StatusBar,
} from 'react-native'

export default function App(){

    const [count, setCount] = useState(0)

    const action = () => {
        setCount(count + 1)
    }

    return(
        <View style={styles.container}>
          <StatusBar backgroundColor={'#000'}/>
            <Text style={styles.labelInfo}>
                App by unit test
            </Text>
            <TouchableOpacity onPress={action} style={styles.button} testID={'buttonCounter'}>
                <Text style={styles.labelButton}>
                    Button
                </Text>
            </TouchableOpacity>
            <Text style={styles.labelCounter} testID={'textLabel'}>
                Count: {count}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#487',
    borderWidth: 10,
    padding: 15,
    alignItems: 'center',
    justifyContent: 'center'
  },
  labelInfo: {
    fontSize: 25,
    color: '#fff',
    marginVertical: 10
  },
  button: {
    paddingVertical: 10,
    paddingHorizontal: 15,
    backgroundColor: '#806',
    marginVertical: 10,
    borderWidth: 1,
    borderRadius: 5
  },
  labelCounter: {
    fontSize: 25,
    color: '#fff',
    padding: 10,
    borderWidth: 2,
    borderRadius: 5,
    marginVertical: 10
  },
  labelButton: {
    color: '#fff',
    fontSize: 23
  }
})