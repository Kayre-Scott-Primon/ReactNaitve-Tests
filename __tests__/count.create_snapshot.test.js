import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer'

/* create a snapshot by defalut and compared to tests */
test('renders correctly', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
})