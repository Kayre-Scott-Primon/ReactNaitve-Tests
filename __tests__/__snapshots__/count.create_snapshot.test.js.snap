// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#487",
      "borderWidth": 10,
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 25,
        "marginVertical": 10,
      }
    }
  >
    App by unit test
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#806",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginVertical": 10,
        "opacity": 1,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
    testID="buttonCounter"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 23,
        }
      }
    >
      Button
    </Text>
  </View>
  <Text
    style={
      Object {
        "borderRadius": 5,
        "borderWidth": 2,
        "color": "#fff",
        "fontSize": 25,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    testID="textLabel"
  >
    Count: 
    0
  </Text>
  <Text
    style={
      Object {
        "borderRadius": 5,
        "borderWidth": 2,
        "color": "#fff",
        "fontSize": 25,
        "marginVertical": 10,
        "padding": 10,
      }
    }
    testID="textLabel"
  >
    Count: 
    0
  </Text>
</View>
`;
